toptal/picasso
# feat: replace Calendar library

[FX-3966]

### Description

This pull request merges the `feature/replace-calendar-library` branch into `master` and fixes a bug in https://github.com/toptal/picasso/pull/3602/commits/4c6708d9ae67bed1046dacb238fba6ae4b8c6f78 (bug was caught during Client Portal tests)

Most of the changes have been already reviewed in previous pull requests:

- https://github.com/toptal/picasso/pull/2996
- https://github.com/toptal/picasso/pull/3584
- https://github.com/toptal/picasso/pull/3579

---

Alpha packages were tested in the following apps (and migration was done along the way):

- Client Portal (https://github.com/toptal/client-portal/pull/6667) tests are ✅ 
- Staff Portal (https://github.com/toptal/staff-portal/pull/10443) tests are ✅ 
- Top Scheduler Frontend (https://github.com/toptal/top-scheduler-frontend/pull/435) tests are ✅ 

### How to test

- All checks should pass
- ClientPortal ([source link for scheduling a break](https://github.com/toptal/client-portal/blob/e7e49b6d032647e701d7195d2fd593ca2adf910f/hosts/client-portal/containers/picassoModals/ScheduleEngagementBreak/Content/index.tsx#L133) and [source link for ending engagement](
https://github.com/toptal/client-portal/blob/a2a43bebb6e8afe0f01b2d7b510d062421050ff5/hosts/client-portal/containers/picassoModals/OffboardingWizardModal/components/EndEngagement/components/EndEngagementContent/EndEngagementForm/index.tsx#L265))
  - visit https://portal-client-portal-6667-fx-3997-make-calendar.toptal.rocks/jobs/VjEtSm9iLTMxNTYwNw
  - log in as `Bruen, Deckow and Tremblay` client
  - schedule a break at https://portal-client-portal-6667-fx-3997-make-calendar.toptal.rocks/jobs/VjEtSm9iLTMxNTYwNw, play with Engagement options, compare with `staging` Client Portal
  - end engagement (follow instructions from https://github.com/toptal/client-portal/pull/5389) at https://portal-client-portal-6667-fx-3997-make-calendar.toptal.rocks/engagements/VjEtRW5nYWdlbWVudC0zMzY4NDM


- TopSchedulerFrontend ([source link](https://github.com/toptal/top-scheduler-frontend/blob/b90f8db9a2f1215152139aae3fdbec8f66976448/src/modules/timeslots/components/DatePicker/DatePicker.tsx#L21))
  - check out https://github.com/toptal/top-scheduler-frontend/pull/435
  - run `yarn storybook`
  - visit http://localhost:6006/?path=/story/timeslots-components-datenavigator--default and http://localhost:6006/?path=/story/timeslots-components-datepicker--default stories


- Staff Portal ([source link](https://github.com/toptal/staff-portal/blob/47241c1aa564ada9213e1fbc7d93da3e04198c99/libs/ui/src/containers/DayOffCalendar/DayOffCalendar.tsx#L42))
  - visit https://staff-portal-pr-10443.toptal.rocks/roles/100010/dayoff and compare it with `staging` Staff Portal

### Development checks

- [N/A] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [N/A] Annotate all `props` in component with documentation
- [N/A] Create `examples` for component
- [N/A] Covered with tests

**Breaking change**

- [N/A] codemod is created and showcased in the changeset

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-3966]: https://toptal-core.atlassian.net/browse/FX-3966?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
